packages/contracts/test/voting/multisig.ts

```diff
@@ -90,119 +95,138 @@ describe('Multisig', function () {
-          minApprovals,
-        })
+        multisig.initialize(dao.address, addresslist(5), multisigSettings)
       ).to.be.revertedWith(ERRORS.ALREADY_INITIALIZED);
     });
+
+    it('adds the initial addresses to the address list', async () => {
+      expect(await multisig.addresslistLength()).to.equal(0);
+
+      multisigSettings.minApprovals = 2;
+      await multisig.initialize(dao.address, addresslist(2), multisigSettings);
+
+      expect(await multisig.addresslistLength()).to.equal(2);
+      expect(await multisig.isListed(signers[0].address)).to.equal(true);
+      expect(await multisig.isListed(signers[1].address)).to.equal(true);
+    });
+
+    it('should set the `minApprovals`', async () => {
+      await multisig.initialize(dao.address, addresslist(5), multisigSettings);
+      expect((await multisig.multisigSettings()).minApprovals).to.be.eq(
+        multisigSettings.minApprovals
+      );
+    });
+
+    it('should set `onlyListed`', async () => {
+      await multisig.initialize(dao.address, addresslist(5), multisigSettings);
+      expect((await multisig.multisigSettings()).onlyListed).to.be.eq(
+        multisigSettings.onlyListed
+      );
+    });
+
+    it('should emit `MultisigSettingsUpdated` during initialization', async () => {
+      await expect(
+        multisig.initialize(dao.address, addresslist(5), multisigSettings)
+      )
+        .to.emit(multisig, MULTISIG_EVENTS.MULTISIG_SETTINGS_UPDATED)
+        .withArgs(multisigSettings.onlyListed, multisigSettings.minApprovals);
+    });
+  });
+
+  describe('updateMultisigSettings:', async () => {
+    beforeEach(async () => {
+      await multisig.initialize(dao.address, addresslist(5), multisigSettings);
+    });
+
+    it('should not allow to set minApprovals larger than the address list length', async () => {
+      let addresslistLength = (await multisig.addresslistLength()).toNumber();
+
+      multisigSettings.minApprovals = addresslistLength + 1;
+
+      await expect(
+        multisig.updateMultisigSettings(multisigSettings)
+      ).to.be.revertedWith(
+        customError(
+          'MinApprovalsOutOfBounds',
+          addresslistLength,
+          multisigSettings.minApprovals
+        )
+      );
+    });
+
+    it('should not allow to set minApprovals would be set to 0', async () => {
+      multisigSettings.minApprovals = 0;
+      await expect(
+        multisig.updateMultisigSettings(multisigSettings)
+      ).to.be.revertedWith(customError('MinApprovalsOutOfBounds', 1, 0));
+    });
+
+    it('should emit MultisigSettingsUpdated when updateMutlsigSettings gets called', async () => {
+      await expect(multisig.updateMultisigSettings(multisigSettings))
+        .to.emit(multisig, MULTISIG_EVENTS.MULTISIG_SETTINGS_UPDATED)
+        .withArgs(multisigSettings.onlyListed, multisigSettings.minApprovals);
+    });
   });
 
-  describe('Addresslisting members: ', async () => {
+  describe('Addresslisting members:', async () => {
```

Comment:
maybe `Addresslist members`

Reply:
Improved in 0dbbd2b6ffc041ae71ed2dd7075a318897f2f9f6